diff -up cups-1.4.1/cgi-bin/admin.c.str3390 cups-1.4.1/cgi-bin/admin.c
--- cups-1.4.1/cgi-bin/admin.c.str3390	2009-08-29 08:05:14.000000000 +0200
+++ cups-1.4.1/cgi-bin/admin.c	2009-10-27 14:44:53.000000000 +0100
@@ -1044,6 +1044,11 @@ do_am_printer(http_t *http,		/* I - HTTP
         if ((attr = ippFindAttribute(oldinfo, "printer-location",
 	                             IPP_TAG_TEXT)) != NULL)
           cgiSetVariable("PRINTER_LOCATION", attr->values[0].string.text);
+
+	if ((attr = ippFindAttribute(oldinfo, "printer-is-shared",
+				     IPP_TAG_BOOLEAN)) != NULL)
+	  cgiSetVariable("PRINTER_IS_SHARED",
+			 attr->values[0].boolean ? "1" : "0");
       }
 
       cgiCopyTemplateLang("modify-printer.tmpl");
